airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py

```diff
@@ -280,6 +277,148 @@ def test_secret_is_properly_marked(self, connector_spec_dict: dict, detailed_log
                     {properties}"""
             )
 
+    def test_property_type_is_not_array(self, connector_spec_dict: dict):
+        """
+        Each field has one or multiple types, but the UI only supports a single type and optionally "null" as a second type.
+        """
+        specification = connector_spec_dict["connectionSpecification"]
+        errors = []
+        for type_path, value in dpath.util.search(specification, "**/properties/*/type", yielded=True):
+            if isinstance(value, List):
+                if len(value) != 2:
+                    errors.append(f"{type_path} is not either a simple type or an array of a simple type plus null: {value}")
```

Comment:
I would suggest adding an example in the error message to show what an array of a simple type plus null is :D I know it sounds obvious, but the more explicit we make the errors the less we receive community solicitations for troubleshooting.